frontend/playground-frontend/src/components/Dashboard/Dashboard.js

```diff
@@ -126,63 +131,147 @@ const formatDate = (unixTime) => {
-              >
-                {row.name}
-              </TableCell>
-              <TableCell component="th" scope="row" className="row-style">
-                {row.type}
-              </TableCell>
-              <TableCell align="left" className="row-style">
-                {row.input}
-              </TableCell>
-              <TableCell align="left" className="row-style">
-                {formatDate(row.date)}
-              </TableCell>
-              <TableCell align="left">
-                <StatusDisplay
-                  statusType={row.statusType}
-                  status={row.status}
-                />
-              </TableCell>
-            </TableRow>
-          ))}
-        </TableBody>
-      </Table>
-    </TableContainer>
+            />
+          </Box>
+          <TableContainer style={{ display: "flex", justifyContent: "center" }}>
+            <Table sx={{ minWidth: 400, m: 2 }}>
+              <TableHead>
+                <TableRow>
+                  <TableCell className="dashboard-header">Name</TableCell>
+                  <TableCell className="dashboard-header">Type</TableCell>
+                  <TableCell className="dashboard-header" align="left">
+                    Date
+                  </TableCell>
+                  <TableCell className="dashboard-header" align="left">
+                    Status
+                  </TableCell>
+                </TableRow>
+              </TableHead>
+              <TableBody>
+                {executiontable.map((row) => (
+                  <TableRow
+                    key={row.execution_id}
+                    sx={{
+                      "&:last-child td, &:last-child th": { border: 0 },
+                      cursor: "pointer",
+                    }}
+                    onClick={() => navigate("/")}
+                    hover
+                  >
+                    <TableCell
+                      component="th"
+                      scope="row"
+                      className="dashboard-header"
+                    >
+                      {row.name}
+                    </TableCell>
+                    <TableCell component="th" scope="row" className="row-style">
+                      {toTitleCase(row.data_source)}
+                    </TableCell>
+                    <TableCell align="left" className="row-style">
+                      {formatDate(new Date(row.timestamp))}
+                    </TableCell>
+                    <TableCell align="left">
+                      <StatusDisplay
+                        statusType={row.status}
+                        status={`${row.progress.toFixed(2)}%`}
+                      />
+                    </TableCell>
+                    <TableCell align="left">
+                      <IconButton
+                        icon="download"
+                        accessibilityLabel={"Download"}
+                        size={"md"}
+                        onClick={async (e) => {
```

Comment:
is it possible to make this code more readable? Declare the onClick function outside and call it here i.e.

```
function handleOnClick() {
// your full function here
}
...
<Icon Button onClick={handleOnClick} />
```

Reply:
100% this suggestion @dwu359